client/src/main/java/org/eclipse/hono/client/HonoClient.java

```diff
@@ -68,6 +68,18 @@
      */
     Future<HonoClient> connect();
 
+    /**
+     * Disconnects from the Hono server.
+     */
+    void disconnect();
+
+    /**
+     * Disconnects from the Hono server.
+     *
+     * @param completionHandler The completion handler to invoke with the result of the operation.
```

Comment:
what are the possible results?

Reply:
the result should indicate whether the disconnect action succeeded or failed.

Reply:
I know that. But the user of this API doesn't and should be informed about it by means of - you already guessed it - the JavaDoc comment ;-)

Reply:
ok, thanks! will fix